### ISSUE TITLE
Can not represent IEEE Infinity

### ISSUE THREAD

**Issue by andygill** [AccelerateHS/accelerate#136]
My program gives:

````
/tmp/accelerate-cuda-17482/dragon17484.cu(23): error: no instance of function template "atomicCAS32" matches the argument list
            argument types are: (Word8 *, Word8, __nv_bool)

1 error detected in the compilation of 

"/tmp/tmpxft_00004456_00000000-6_dragon17484.cpp1.ii".
ecc-ldpc-accelerate: nvcc terminated abnormally (2)
/tmp/accelerate-cuda-17482/dragon17486.cu(21): error: identifier "Infinity" is undefined

1 error detected in the compilation of "/tmp/tmpxft_0000445f_00000000-6_dragon17486.cpp1.ii".
ecc-ldpc-accelerate: nvcc terminated abnormally (2)
/tmp/accelerate-cuda-17482/dragon17485.cu(9): error: identifier "Infinity" is undefined

1 error detected in the compilation of "/tmp/tmpxft_0000445a_00000000-6_dragon17485.cpp1.ii".
ecc-ldpc-accelerate: nvcc terminated abnormally (2)
````

I suspect that it is this expression

````
lift (1/0::Double,1/0::Double)
```

I'll try get a smaller sample program.

**Comment by mbudde** [comment]
I believe that this issue has been fixed in AccelerateHS/accelerate-cuda@5d4baaaf3364eba0b79bb24e4a14d134440e70ef.

**Comment by tmcdonell** [comment]
Yes, well spotted. Thanks!
